packages/itwinui-react/src/core/Tabs/Tabs.tsx

```diff
@@ -526,85 +503,399 @@ export const Tabs = (props: TabsProps) => {
+        {children}
+      </Box>
+    );
+  } else {
+    return <></>;
+  }
+}) as PolymorphicForwardRefComponent<'div'>;
+TabsTabDescription.displayName = 'Tabs.TabDescription';
+
+// ----------------------------------------------------------------------------
+// Tabs.Actions component
+
+const TabsActions = React.forwardRef((props, ref) => {
+  const { className, children, ...rest } = props;
+
+  const { type } = useSafeContext(TabsContext);
+
+  if (type !== 'pill') {
+    return (
       <Box
-        as='ul'
-        className={cx(
-          'iui-tabs',
-          `iui-${type}`,
-          {
-            'iui-green': color === 'green',
-            'iui-animated': type !== 'default' && isClient,
-            'iui-not-animated': type !== 'default' && !isClient,
-            'iui-large': hasSublabel,
-          },
-          tabsClassName,
-        )}
-        data-iui-overflow={overflowOptions?.useOverflow}
-        data-iui-scroll-placement={scrollingPlacement}
-        role='tablist'
-        ref={refs}
-        onKeyDown={onKeyDown}
+        className={cx('iui-tabs-actions-wrapper', className)}
+        ref={ref}
         {...rest}
       >
-        {labels.map((label, index) => createTab(label, index))}
+        {children}
       </Box>
+    );
+  } else {
+    return <></>;
+  }
+}) as PolymorphicForwardRefComponent<'div'>;
+TabsActions.displayName = 'Tabs.Actions';
 
-      {actions && (
-        <Box className='iui-tabs-actions-wrapper'>
-          <Box className='iui-tabs-actions'>{actions}</Box>
-        </Box>
-      )}
+// ----------------------------------------------------------------------------
+// Tabs.Action component
 
-      {children && (
-        <Box
-          className={cx('iui-tabs-content', contentClassName)}
-          role='tabpanel'
-        >
-          {children}
-        </Box>
-      )}
-    </Box>
+const TabsAction = React.forwardRef((props, ref) => {
+  const { className, children, ...rest } = props;
+
+  const { type } = useSafeContext(TabsContext);
+
+  if (type !== 'pill') {
+    return (
+      <Box className={cx('iui-tabs-actions', className)} ref={ref} {...rest}>
+        {children}
+      </Box>
+    );
+  } else {
+    return <></>;
+  }
+}) as PolymorphicForwardRefComponent<'div'>;
+TabsAction.displayName = 'Tabs.Action';
+
+// ----------------------------------------------------------------------------
+// Tabs.Panels component
+
+const TabsPanels = ({
+  children,
+}: {
+  children: React.ReactNode[] | React.ReactNode;
+}) => {
+  const { currentActiveIndex } = useSafeContext(TabsContext);
+
+  const items = Array.isArray(children) ? children : [children];
+
+  return (
+    <>
+      {items.map((item, index) => {
+        return (
+          <TabsContext.Provider
+            value={{ index, currentActiveIndex }}
+            key={index}
+          >
+            {item}
+          </TabsContext.Provider>
+        );
+      })}
+    </>
   );
 };
+TabsPanels.displayName = 'Tabs.Panels';
```

Comment:
This is for content within the Tab right? I think `Tabs.Content` would be more a informative name